.github/workflows/windows.yml

```diff
@@ -0,0 +1,93 @@
+name: mingw-w64
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  mingw-w64:
+    name: Build and test
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Git branch
+        uses: actions/checkout@v2
+
+      - name: Cache Msys2
+        id: cache-msys2
+        uses: actions/cache@v1.1.0
+        with:
+          path: msys64
+          key: ${{ runner.os }}-msys64
+
+      - name: Get Msys2
+        if: steps.cache-msys2.outputs.cache-hit != 'true'
+        run: |
+          Invoke-WebRequest -Outfile msys2.tar.xz -Uri "https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.tar.xz"
+          xz -d msys2.tar.xz
+          tar -xf msys2.tar
+          rm msys2.tar
+
+      - name: Add Msys2 to Path
+        run: |
+          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/mingw64/bin"
+          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/usr/bin"
+
+      - shell: cmd
+        if: steps.cache-msys2.outputs.cache-hit != 'true'
+        run: bash -lc 'exit'
+
+      - name: Update Pacman
+        run: pacman -Syy --needed --ask=20 --noconfirm
+
+      - name: Cache pacman
+        id: cache-pacman
+        uses: actions/cache@v1.1.0
+        with:
+          path: msys64/var/cache/pacman/pkg
+          key: ${{ runner.os }}-pacman-${{ hashFiles('msys64/var/lib/pacman/sync/mingw64.db') }}
+          restore-keys: |
+            ${{ runner.os }}-pacman-
+
+      - name: Update Msys2 Packages
+        run: pacman -Suu --needed --ask=20 --noconfirm
```

Comment:
Probably it's me that is missing something, currently cloning pacman source so I figure out more about it.

But yeah, I added double u to just in case there's a downgrade because of bugs or incompatibilities since that has happened before

Reply:
In lib/libalpm/sync.c

```c
		if(enable_downgrade) {
			/* check IgnorePkg/IgnoreGroup */
			if(alpm_pkg_should_ignore(handle, spkg)
					|| alpm_pkg_should_ignore(handle, lpkg)) {
				_alpm_log(handle, ALPM_LOG_WARNING, _("%s: ignoring package downgrade (%s => %s)\n"),
						lpkg->name, lpkg->version, spkg->version);
			} else {
				_alpm_log(handle, ALPM_LOG_WARNING, _("%s: downgrading from version %s to version %s\n"),
						lpkg->name, lpkg->version, spkg->version);
				return 1;
			}
		} else {
			alpm_db_t *sdb = alpm_pkg_get_db(spkg);
			_alpm_log(handle, ALPM_LOG_WARNING, _("%s: local (%s) is newer than %s (%s)\n"),
					lpkg->name, lpkg->version, sdb->treename, spkg->version);
		}
```

I think they ignore the ask option since they don't have anything with `config->ask` (where they store the `(unsigned int)20` into) nearby or in any of the functions

(It's so hard to read someone else's code from scratch.)